src/hooks/screen/useCreatePostHook.js

```diff
@@ -12,27 +12,22 @@
 const useCreatePostHook = (isAnonymous) => {
   const {params = {}} = useRoute();
   const {topic} = params;
+  const [selectedTopic, setSelectedTopic] = React.useState(topic);
   const [anonUserInfo, setAnonUserInfo] = React.useState(null);
+  const [headerTitle, setHeaderTitle] = React.useState(
+    selectedTopic ? `Create Post in #${selectedTopic}` : 'Create Post'
+  );
 
-  const headerTitle = topic ? `Create Post in #${topic}` : 'Create Post';
-
-  const isInCreatePostTopicScreen = !!topic;
+  const isInCreatePostTopicScreen = !!selectedTopic;
 
   const getAnonUserInfo = React.useCallback(async () => {
     try {
       const response = await AnonUserInfoRepo.getPostAnonUserInfo();
-      if (response.isSuccess) {
-        setAnonUserInfo(response.data);
-      } else {
-        showMessage({
-          message: response.error,
-          type: 'danger'
-        });
-      }
+      setAnonUserInfo(response.data);
     } catch (e) {
       console.log('Error in getAnonUserInfo', e);
       showMessage({
-        message: e,
+        message: JSON.stringify(e),
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

Error handling here could be improved. Instead of logging the error and showing a message with the entire error object stringified, it would be better to display a user-friendly error message and log the specific error details separately. Also, consider using a more robust logging solution than `console.log` for better tracking and debugging.

```diff
- console.log('Error in getAnonUserInfo', e);
- showMessage({
-  message: JSON.stringify(e),
-  type: 'danger'
- });
+ console.error('Error in getAnonUserInfo', e);
+ showMessage({
+  message: "There was an issue fetching anonymous user info. Please try again later.",
+  type: 'danger'
+ });
```



<!-- This is an auto-generated comment by OSS CodeRabbit -->